modules/shared/components/atoms/TextInputs/TextInput.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import type { FC, ReactElement } from 'react';
+import className from 'classnames';
+import DeleteIcon from '../../icons/inputFocus.svg';
+import ErrorIcon from '../../icons/inputError.svg';
+import SuccessIcon from '../../icons/inputSuccess.svg';
+import ArrowDownIcon from '../../icons/inputArrowDown.svg';
+import HomeIcon from '../../icons/inputHome.svg';
+import styles from './TextInput.module.css';
+import type { ITextInputs } from './types/ITextInputs';
+import * as ETextInput from './types/ETextInput';
+
+const TextInput: FC<ITextInputs.IProps> = ({
+  label,
+  id,
+  variants,
+  inputType,
+  disabled,
+  letter,
+  ...props
+}): ReactElement => {
+  const [inputVal, setInputVal] = React.useState<string>('');
+
+  const changeHandler = (e: React.FormEvent<HTMLInputElement>): void => {
+    setInputVal(e.currentTarget.value);
+  };
+
+  const hideIconHandler = (): void => {
+    setInputVal('');
+  };
+
+  const inputClasses: string = className(
+    styles['form-input'],
+    {
+      [styles.error]: variants === ETextInput.Variants.Error && !disabled,
+      [styles.success]: variants === ETextInput.Variants.Success && !disabled,
+    },
+    {
+      [styles['input-with-right-icon']]:
+        inputType === ETextInput.InputType.RightIcon,
+      [styles['input-with-left-icon']]:
+        inputType === ETextInput.InputType.LeftIcon,
+      [styles['input-with-prefix']]: inputType === ETextInput.InputType.Prefix,
+      [styles['input-with-dropdown-prefix']]:
+        inputType === ETextInput.InputType.PrefixDropdown,
+      [styles['input-with-choices']]:
+        inputType === ETextInput.InputType.Choices,
+    },
+  );
+
+  /*
+    when it comes to authentication we need to validate
+  */
+  // const inputId = id.trim().split(' ').join('');
+
+  return (
+    <div className={styles['form-group']}>
+      {label !== undefined && (
```

Comment:
this condition doesn't work like it should 
if there is no label giving it stills have a label element rendered
![Screenshot from 2021-05-16 23-28-13](https://user-images.githubusercontent.com/68826445/118413224-74775e00-b69e-11eb-8b1a-f135c202e5d7.png)